### PR TITLE
doesn't add empty CATEGORIES if field empty in interactive mode

### PR DESCRIPTION
**Pull request by titibandit**
fixes issue #1300

### CODE REVIEW

File: khal/khalendar/event.py [hunk 1]
```diff
@@ -524,8 +524,9 @@ def categories(self) -> str:
 
     def update_categories(self, categories: List[str]) -> None:
         assert isinstance(categories, list)
+        categories = [c.strip().lower() for c in categories if c != ""]
```
Comment (WhyNotHugo) [inline khal/khalendar/event.py:527]: `.lower()` is not correct here; categories may be uppercase or mixed case.

File: khal/khalendar/event.py [hunk 2]
```diff
@@ -524,8 +524,9 @@ def categories(self) -> str:
...
         self._vevents[self.ref].pop('CATEGORIES', False)
-        if categories:
+        if len(categories) > 0:
```
Comment (WhyNotHugo) [inline khal/khalendar/event.py:529]: Does this make any difference?

**Review by WhyNotHugo — CHANGES_REQUESTED** [review]

File: khal/khalendar/event.py (hunk 1 above)
Comment (titibandit) [reply khal/khalendar/event.py]: That's right. I removed it.

File: khal/khalendar/event.py (hunk 2 above)
Comment (titibandit) [reply khal/khalendar/event.py]: It doesn't :) changed accordingly

**Comment by WhyNotHugo** [comment]
I tested this with the following sequence (with `ikhal`):

- Edit an event and add a category to it.
- Check the file (it has `CATEGORIES:test`).
- Edit the event again and remove the category.
- `CATEGORIES:` is gone.

**Review by WhyNotHugo — APPROVED** [review]
LGTM

**Pull request merged**